### PR TITLE
search tweaks

### PR DESCRIPTION
**Pull request by chrisnojima**
- [x] In search results: service icons on the right side when searching Keybase should be the colored ones, not the monochrome/icon font ones.
- [x] In search results if you select a non-keybase service and search if you see a result that has keybase info, if you track them the username should be green (not orange)
- [x] In search results: Rows cover up border / scrollbar: this is due to https://github.com/keybase/client/blob/master/shared/search/user-search/render.desktop.js#L115 the background being set to white causes this overlap
- [x] In search results: Disallow finding yourself
- [x] In search results: search input should be auto focused. Keep focus on input when changing the service type.
- [x] In search results: The services / input shouldn't scroll, just the results
- [x] In search area hint text should be left aligned and be aligned with actual input text (currently it floats above the actual typed text)
- [x] In search area hint text should be 'Search a xxxxx user' (with no group) and 'Add a xxxx user' when a group exists
- [x] In search area unselected services should be opacity 0.6. on hover/selected 1.0 opacity.
- [x] In search area removing the query should show the group or initial screen (currently shows no results)
- [x] In group view, selected user should remain selected (not just on hover). So if they're shown on the right pane, they should be highlighted blue
- [x] In group view actions should be clickable links, not row items (no highlight on hover, etc)
- [x] In search group view usersnames should be semibold 
- [x] In group view the remove 'x' button should be black_60 on hover, no animations
- [x] In group view we need the 'Clear search' link on the top right, see https://zpl.io/ZDWKbY
- [x] Main nav search needs hand pointer on hover
- [x] Profile card: Trakcers / Tracking links should be black_60 style not black_40 (maybe text style is wrong)
- [x] Profile card: Send invite link should be underlined on hover
- [x] Remove dumb-only copy of components
- [ ] In search results show spinner while waiting for results : TODO waiting to rebase to get icon stuff. gifs shouldn't do srcset

### CODE REVIEW

**Comment by keybase-ci-visdiff** [comment]
:mag_right: The commits 0675cf484270...82840aab34c2 introduced some visual changes on win32:
 * Changed: **Request_Invite-Name_Error** [(before)](https://keybase-app-visdiff.s3.amazonaws.com/1468266535847-0675cf484270-82840aab34c2-win32/Request_Invite-Name_Error-0675cf484270.png) [(after)](https://keybase-app-visdiff.s3.amazonaws.com/1468266535847-0675cf484270-82840aab34c2-win32/Request_Invite-Name_Error-82840aab34c2.png)  
   Request_Invite-Name_Error comparison

**Comment by keybase-ci-visdiff** [comment]
:mag_right: The commits 4c1f0cff4fb3...55f94a8bb101 introduced some visual changes on win32:
 * Changed: **user_search-Normal** [(before)](https://keybase-app-visdiff.s3.amazonaws.com/1468269584403-4c1f0cff4fb3-55f94a8bb101-win32/user_search-Normal-4c1f0cff4fb3.png) [(after)](https://keybase-app-visdiff.s3.amazonaws.com/1468269584403-4c1f0cff4fb3-55f94a8bb101-win32/user_search-Normal-55f94a8bb101.png)  
   user_search-Normal comparison

**Comment by keybase-ci-visdiff** [comment]
:mag_right: The commits 4c1f0cff4fb3...55f94a8bb101 introduced some visual changes on linux:
 * Changed: **user_search-Normal** [(before)](https://keybase-app-visdiff.s3.amazonaws.com/1468269673524-4c1f0cff4fb3-55f94a8bb101-linux/user_search-Normal-4c1f0cff4fb3.png) [(after)](https://keybase-app-visdiff.s3.amazonaws.com/1468269673524-4c1f0cff4fb3-55f94a8bb101-linux/user_search-Normal-55f94a8bb101.png)  
   user_search-Normal comparison

**Comment by keybase-ci-visdiff** [comment]
:mag_right: The commits 32560b07b04f...da45e65404f5 introduced some visual changes on win32:
 * Changed: **unlock_folders-Success** [(before)](https://keybase-app-visdiff.s3.amazonaws.com/1468272364275-32560b07b04f-da45e65404f5-win32/unlock_folders-Success-32560b07b04f.png) [(after)](https://keybase-app-visdiff.s3.amazonaws.com/1468272364275-32560b07b04f-da45e65404f5-win32/unlock_folders-Success-da45e65404f5.png)  
   unlock_folders-Success comparison
 * Changed: **unlock_folders-Single** [(before)](https://keybase-app-visdiff.s3.amazonaws.com/1468272364275-32560b07b04f-da45e65404f5-win32/unlock_folders-Single-32560b07b04f.png) [(after)](https://keybase-app-visdiff.s3.amazonaws.com/1468272364275-32560b07b04f-da45e65404f5-win32/unlock_folders-Single-da45e65404f5.png)  
   unlock_folders-Single comparison

**Comment by keybase-ci-visdiff** [comment]
:mag_right: The commits da45e65404f5...1b9a9b1e852b introduced some visual changes on win32:
 * Changed: **user_search-Normal** [(before)](https://keybase-app-visdiff.s3.amazonaws.com/1468272529026-da45e65404f5-1b9a9b1e852b-win32/user_search-Normal-da45e65404f5.png) [(after)](https://keybase-app-visdiff.s3.amazonaws.com/1468272529026-da45e65404f5-1b9a9b1e852b-win32/user_search-Normal-1b9a9b1e852b.png)  
   user_search-Normal comparison

**Comment by keybase-ci-visdiff** [comment]
:mag_right: The commits da45e65404f5...1b9a9b1e852b introduced some visual changes on linux:
 * Changed: **user_search-Normal** [(before)](https://keybase-app-visdiff.s3.amazonaws.com/1468272648031-da45e65404f5-1b9a9b1e852b-linux/user_search-Normal-da45e65404f5.png) [(after)](https://keybase-app-visdiff.s3.amazonaws.com/1468272648031-da45e65404f5-1b9a9b1e852b-linux/user_search-Normal-1b9a9b1e852b.png)  
   user_search-Normal comparison

**Comment by keybase-ci-visdiff** [comment]
:mag_right: The commits 54e3098cd1cb...1d4595a3f00a introduced some visual changes on win32:
 * Changed: **user_search-Normal** [(before)](https://keybase-app-visdiff.s3.amazonaws.com/1468333899560-54e3098cd1cb-1d4595a3f00a-win32/user_search-Normal-54e3098cd1cb.png) [(after)](https://keybase-app-visdiff.s3.amazonaws.com/1468333899560-54e3098cd1cb-1d4595a3f00a-win32/user_search-Normal-1d4595a3f00a.png)  
   user_search-Normal comparison

**Comment by keybase-ci-visdiff** [comment]
:mag_right: The commits 54e3098cd1cb...1d4595a3f00a introduced some visual changes on linux:
 * Changed: **user_search-Normal** [(before)](https://keybase-app-visdiff.s3.amazonaws.com/1468334039434-54e3098cd1cb-1d4595a3f00a-linux/user_search-Normal-54e3098cd1cb.png) [(after)](https://keybase-app-visdiff.s3.amazonaws.com/1468334039434-54e3098cd1cb-1d4595a3f00a-linux/user_search-Normal-1d4595a3f00a.png)  
   user_search-Normal comparison

**Comment by keybase-ci-visdiff** [comment]
:mag_right: The commits 1d4595a3f00a...e5c5bdb7b5bb introduced some visual changes on win32:
 * Changed: **user_group-Chat_enabled** [(before)](https://keybase-app-visdiff.s3.amazonaws.com/1468337548083-1d4595a3f00a-e5c5bdb7b5bb-win32/user_group-Chat_enabled-1d4595a3f00a.png) [(after)](https://keybase-app-visdiff.s3.amazonaws.com/1468337548083-1d4595a3f00a-e5c5bdb7b5bb-win32/user_group-Chat_enabled-e5c5bdb7b5bb.png)  
   user_group-Chat_enabled comparison

**Comment by keybase-ci-visdiff** [comment]
:mag_right: The commits 1d4595a3f00a...e5c5bdb7b5bb introduced some visual changes on linux:
 * Changed: **user_group-Chat_enabled** [(before)](https://keybase-app-visdiff.s3.amazonaws.com/1468337723564-1d4595a3f00a-e5c5bdb7b5bb-linux/user_group-Chat_enabled-1d4595a3f00a.png) [(after)](https://keybase-app-visdiff.s3.amazonaws.com/1468337723564-1d4595a3f00a-e5c5bdb7b5bb-linux/user_group-Chat_enabled-e5c5bdb7b5bb.png)  
   user_group-Chat_enabled comparison

**Comment by keybase-ci-visdiff** [comment]
:mag_right: The commits e5c5bdb7b5bb...ffdf3e12075c introduced some visual changes on win32:
 * Changed: **user_group-Chat_enabled** [(before)](https://keybase-app-visdiff.s3.amazonaws.com/1468338282044-e5c5bdb7b5bb-ffdf3e12075c-win32/user_group-Chat_enabled-e5c5bdb7b5bb.png) [(after)](https://keybase-app-visdiff.s3.amazonaws.com/1468338282044-e5c5bdb7b5bb-ffdf3e12075c-win32/user_group-Chat_enabled-ffdf3e12075c.png)  
   user_group-Chat_enabled comparison

**Comment by keybase-ci-visdiff** [comment]
:mag_right: The commits e5c5bdb7b5bb...ffdf3e12075c introduced some visual changes on linux:
 * Changed: **user_group-Chat_enabled** [(before)](https://keybase-app-visdiff.s3.amazonaws.com/1468338416001-e5c5bdb7b5bb-ffdf3e12075c-linux/user_group-Chat_enabled-e5c5bdb7b5bb.png) [(after)](https://keybase-app-visdiff.s3.amazonaws.com/1468338416001-e5c5bdb7b5bb-ffdf3e12075c-linux/user_group-Chat_enabled-ffdf3e12075c.png)  
   user_group-Chat_enabled comparison

File: shared/common-adapters/icon.desktop.js [hunk 1]
```diff
@@ -33,11 +33,16 @@ export default class Icon extends Component {
     const fontSizeHint = shared.fontSize(iconType)
 
     if (isFontIcon) {
+      const cleanStyle = {...this.props.style}
+      // We have to blow these styles away else FontIcon gets confused and will overwrite what it calculates
```
Comment (chrisnojima) [inline shared/common-adapters/icon.desktop.js]: https://github.com/callemall/material-ui/issues/4692

**Comment by keybase-ci-visdiff** [comment]
:mag_right: The commits 8b6df0b9ef94...b327d904cbae introduced some visual changes on win32:
 * Changed: **user_search-Normal** [(before)](https://keybase-app-visdiff.s3.amazonaws.com/1468345208272-8b6df0b9ef94-b327d904cbae-win32/user_search-Normal-8b6df0b9ef94.png) [(after)](https://keybase-app-visdiff.s3.amazonaws.com/1468345208272-8b6df0b9ef94-b327d904cbae-win32/user_search-Normal-b327d904cbae.png)  
   user_search-Normal comparison

**Comment by keybase-ci-visdiff** [comment]
:mag_right: The commits 8b6df0b9ef94...b327d904cbae introduced some visual changes on linux:
 * Changed: **user_search-Normal** [(before)](https://keybase-app-visdiff.s3.amazonaws.com/1468345493373-8b6df0b9ef94-b327d904cbae-linux/user_search-Normal-8b6df0b9ef94.png) [(after)](https://keybase-app-visdiff.s3.amazonaws.com/1468345493373-8b6df0b9ef94-b327d904cbae-linux/user_search-Normal-b327d904cbae.png)  
   user_search-Normal comparison

**Comment by chrisnojima** [comment]
@keybase/react-hackers this is ready for review. before i squish / rebase master to get the icon stuff i'll get feedback first since the commits are small and sensible now. when i get a thumb i'll do the rebase / waiting icon

File: shared/common-adapters/clickable-box.js.flow [hunk 2]
```diff
@@ -5,8 +5,8 @@ import React, {Component} from 'react'
 export type Props = {
   style?: Object,
   onClick: () => void,
-  hoverColor?: string,
-  backgroundColor?: string,
+  hoverColor?: ?string,
```
Comment (MarcoPolo) [inline shared/common-adapters/clickable-box.js.flow]: Did this cause a problem?

File: shared/reducers/search.js [hunk 3]
```diff
@@ -11,24 +11,34 @@ export type State = {
...
   results: Array<SearchResult>,
   requestTimestamp: ?Date,
   selectedUsers: Array<SearchResult>,
   userForInfoPane: ?SearchResult,
   showUserGroup: boolean,
+  waiting: boolean,
 }
 
+const searchHintText = (searchPlatform: SearchPlatforms, selectedUsers: Array<SearchResult>): string => (
+  `${selectedUsers.length ? 'Add' : 'Search'} a ${searchPlatform} user`
```
Comment (MarcoPolo) [inline shared/reducers/search.js]: Is that the correct way of saying that? `Search a reddit user` ?

**Comment by MarcoPolo** [comment]
Looks great! Thanks for cleaning this up!

Re: the `UserPane` smart component, I still think that's a good idea. It makes the parent component have to worry about much less, it just uses the component and the rest is taken care of.

I can see the argument that we lose the ability to render a full dumb component of what the user is actually seeing. How about this: If a dumb component takes a smart component (like `search/render.desktop` uses `UserPane`) then you make that a prop and it's passed in.

Our typed connector logic will make it so you have to pass in the correct props, and for dumb mocks you can reuse the dumb mocks of just the user pane passed into the dumb mock of `search/render`.

The parent smart component passes just the `UserPane` smart component.

Gives you the ability to make full screen dumb mocks, and also lets you keep things simple by making more specific smart components.

File: shared/common-adapters/clickable-box.js.flow (hunk 2 above)
Comment (chrisnojima) [inline shared/common-adapters/clickable-box.js.flow]: yeah, i want to dynamically set this, like {hoverColor: this.state.something ? 'value' : null} but it won't let me do that without allowing ?string

File: shared/reducers/search.js (hunk 3 above)
Comment (chrisnojima) [inline shared/reducers/search.js]: its what the comps say @cecileboucheron @malgorithms

**Comment by chrisnojima** [comment]
@MarcoPolo instead of allowing the component to be injected (i think thats a little bit of an abstraction violation, instead i went with allowing ownProps to have the selected user (really this thing is driven by one input anyways). i think this works nicely. if you don't init it like
```<UserPane userForInfoPane={whatever}/>```
then you get the totally connected one.

**Comment by chrisnojima** [comment]
ok i'm going to squish / rebase

**Comment by chrisnojima** [comment]
Replaced by squish/rebased version here: https://github.com/keybase/client/pull/3444

File: shared/reducers/search.js (hunk 3 above)
Comment (MarcoPolo) [inline shared/reducers/search.js]: feels awkward, maybe `Search for a reddit user`?

**Pull request closed without merging**